vng/vng_test.go

```diff
@@ -0,0 +1,216 @@
+package vng_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+	"testing"
+
+	"github.com/spf13/afero"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/pkg/nano"
+	"github.com/brimdata/zed/zio/vngio"
+	"github.com/brimdata/zed/zson"
+)
+
+func FuzzVngRoundtrip(f *testing.F) {
+	f.Fuzz(func(t *testing.T, b []byte) {
+		valuesIn := genValues(bytes.NewReader(b))
+
+		var Fs = afero.NewMemMapFs()
+
+		// Write
+		file, err := Fs.Create("test.vng")
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+		writer, err := vngio.NewWriter(file, vngio.WriterOpts{ColumnThresh: vngio.DefaultColumnThresh, SkewThresh: vngio.DefaultSkewThresh})
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+		for i := range valuesIn {
+			err := writer.Write(&valuesIn[i])
+			if err != nil {
+				t.Errorf("%v", err)
+			}
+		}
+		err = writer.Close()
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+
+		// Read
+		file, err = Fs.Open("test.vng")
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+		context := zed.NewContext()
+		reader, err := vngio.NewReader(context, file)
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+		valuesOut := make([]zed.Value, 0, len(valuesIn))
+		for {
+			value, err := reader.Read()
+			if err != nil {
+				t.Errorf("%v", err)
+			}
+			if value == nil {
+				break
+			}
+			valuesOut = append(valuesOut, *value)
```

Comment:
Is this a UAF of some buffer in the reader? Maybe I need to copy the underlying bytes somewhere?

Reply:
Yep :S